crates/db_schema/src/impls/local_user.rs

```diff
@@ -215,6 +215,35 @@ impl LocalUser {
       blocked_instances,
     })
   }
+
+  /// Checks to make sure the acting moderator is higher than the target moderator
+  pub async fn is_higher_admin_check(
+    pool: &mut DbPool<'_>,
+    admin_person_id: PersonId,
+    target_person_ids: &[PersonId],
+  ) -> Result<bool, Error> {
+    let conn = &mut get_conn(pool).await?;
+
+    // Build the list of persons
+    let mut persons = target_person_ids.to_owned();
+    persons.push(admin_person_id);
+    persons.dedup();
+
+    let res = local_user::table
+      .filter(local_user::admin.eq(true))
+      .filter(local_user::person_id.eq_any(persons))
+      .order_by(local_user::id)
```

Comment:
This is the main weirdness of this PR, but it might not be an issue.

Unlike the `community_moderator` table, the `local_user` table doesn't have a `published` column, so I can't sort by it. So the admin priority list is simply by the `local_user.id`

Its probably not an issue, because we also determine top admin by the first id.

Reply:
Later, this can be fixed by replacing `admin` with a field called `became_admin` with the type `Nullable<Timestamptz>`.

The migration can set it to `person.published` for existing admins. The only case in which this changes the order is if there were concurrent insertions to `local_user` in which one insertion started first but did not generate the local user id first.

Reply:
Leave a comment to make this explicit. And I believe it would be fixed by https://github.com/LemmyNet/lemmy/issues/3703 so you can link that.